packages/@aws-containers/aws-ecs-builder/lib/extensions/appmesh.ts

```diff
@@ -0,0 +1,326 @@
+        // Name the namespace after the environment name.
+        // Service DNS will be like <service id>.<environment id>
+        name: this.parentService.environment.id,
+      });
+    }
+  }
+
+  public modifyTaskDefinitionProps(props: ecs.TaskDefinitionProps) {
+    // Find the app extension, to get its port
+    const containerextension = this.parentService.serviceDescription.get('service-container') as Container;
+
+    if (!containerextension) {
+      throw new Error('Appmesh extension requires an application extension');
+    }
+
+    return {
+      ...props,
+
+      // App Mesh requires AWS VPC networking mode so that each
+      // task can have its own IP address
+      networkMode: ecs.NetworkMode.AWS_VPC,
+
+      // This configures the envoy container as a proxy for all
+      // traffic going into and out of the task, with a few exceptions
+      // for metadata endpoints or other ports that need direct
+      // communication
+      proxyConfiguration: new ecs.AppMeshProxyConfiguration({
+        containerName: 'envoy',
+        properties: {
+          appPorts: [containerextension.trafficPort],
+          proxyEgressPort: 15001,
+          proxyIngressPort: 15000,
+
+          // The App Mesh proxy runs with this user ID, and this keeps its
+          // own outbound connections from recursively attempting to infinitely proxy.
+          ignoredUID: 1337,
+
+          // This GID is ignored and any outbound traffic originating from containers that
+          // use this group ID will be ignored by the proxy. This is primarily utilized by
+          // the FireLens extension, so that outbound application logs don't have to go through Envoy
+          // and therefore add extra burden to the proxy sidecar. Instead the logs can go directly
+          // to CloudWatch
+          ignoredGID: 1338,
+
+          egressIgnoredIPs: [
+            '169.254.170.2', // Allow services to talk directly to ECS metadata endpoints
+            '169.254.169.254', // and EC2 instance endpoint
+          ],
+
+          // If there is outbound traffic to specific ports that you want to
+          // ignore the proxy those ports can be added here.
+          egressIgnoredPorts: [],
+        },
+      }),
+    } as ecs.TaskDefinitionProps;
+  }
+
+  public useTaskDefinition(taskDefinition: ecs.TaskDefinition) {
+    var region = cdk.Stack.of(this.scope).region;
+    var appMeshRepo;
+    var ownerAccount;
+
+    // This is currently necessary because App Mesh has different images in each region,
+    // and some regions have their images in a different account. See:
+    // https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+    if (region === 'me-south-1') {
+      ownerAccount = APPMESH_ECR_ACCOUNTS.ME_SOUTH_1;
+    } else if (region === 'ap-east-1') {
+      ownerAccount = APPMESH_ECR_ACCOUNTS.AP_EAST_1;
+    } else {
+      ownerAccount = APPMESH_ECR_ACCOUNTS.DEFAULT;
+    }
```

Comment:
A valiant effort, but unfortunately it won't really work for environment-agnostic stacks. What you probably ned here is a CloudFormation Mapping. Check out how we handle a very similar usecase in CodeBuild: https://github.com/aws/aws-cdk/blob/d45a57c22a006f682e584c5ef6c8ef3f416caf86/packages/%40aws-cdk/aws-codebuild/lib/linux-gpu-build-image.ts#L98-L114

Reply:
Thanks! I adjusted this to use a CloudFormation map. I do have a future note to add something to dedupe the mapping. Right now it is creating a separate but identical map each time for each instantiation of the extension, and ideally it would have all extensions making use of one global, shared mapping. This will be improved in the future.